File: Editorial/0101-0200/0111_Minimum_Depth_of_Binary_Tree/C/1_Depth_First_Search.c
// C
int dfs(struct TreeNode* root) {
    if (!root) {
        return 0;
    }
    // If only one of child is non-null, then go into that recursion.
    if (!root->left) {
        return 1 + dfs(root->right);
    } else if (!root->right) {
        return 1 + dfs(root->left);
    }
    // Both children are non-null, hence call for both children.
    int left_depth = dfs(root->left);
    int right_depth = dfs(root->right);
    return 1 + (left_depth < right_depth ? left_depth : right_depth);
}
int minDepth(struct TreeNode* root) { return dfs(root); }

File: Editorial/0101-0200/0111_Minimum_Depth_of_Binary_Tree/CSharp/1_Depth_First_Search.cs
// C#

public class Solution {
    private int Dfs(TreeNode root) {
        if (root == null) {
            return 0;
        }

        // If only one of child is non-null, then go into that recursion.
        if (root.left == null) {
            return 1 + Dfs(root.right);
        } else if (root.right == null) {
            return 1 + Dfs(root.left);
        }

        // Both children are non-null, hence call for both children.
        return 1 + Math.Min(Dfs(root.left), Dfs(root.right));
    }

    public int MinDepth(TreeNode root) {
        return Dfs(root);
    }
}
